File: AuthorizationService/Tests/test_crypto.py
from SRP_module import SRP

srp = SRP()

original = "Hello word!"

encode_str = srp.encrypt_by_key(original, str(108969))
decode_str = srp.decrypt_by_key(encode_str, str(108969))


print(original)
print(encode_str)
print(decode_str)

assert original == decode_str

File: APIGateway/test/test_crypto.cpp
#include <iostream>
#include <string>
#include "../include/SRP_module.hpp"
#include <cassert>

int main()
{
    SRP srp;

    std::string original = "Hello, world!";
    std::string key = "216345";

    std::string encrypted = srp.encrypt_by_key(original, key);
    std::string decrypted = srp.encrypt_by_key(encrypted, key);

    std::cout << "Original: " << original << std::endl;
    std::cout << "Encrypted: " << encrypted << std::endl;
    std::cout << "Decrypted: " << decrypted << std::endl;

    assert(original.compare(decrypted) == 0);


    return 0;
}
